packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -313,15 +318,37 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         return (minIdx, minVotes);
     }
 
-    // solhint-disable-next-line no-empty-blocks
     function resolveElection() external virtual override {
-        // TODO Check preconditions
-        // TODO Compare lists and -> fill TO_REMOVE, and TO_ADD
-        // TODO Move votes from TO_REMOVE to TO_ADD
-        // TODO Burn missing TO_REMOVE
-        // TODO Mint missing TO_ADD
-        // TODO Flip epochs
-        // TODO _electionStore().latestElectionDataIndex++;
+        ElectionStore storage store = _electionStore();
+        MemberToken memberToken = MemberToken(store.memberTokenAddress);
+        address[] memory nextEpochMembers = _currentElectionData().nextEpochMembers;
+        uint currentMembersSize = store.members.length;
+        uint nextEpochMembersSize = nextEpochMembers.length;
+
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (!isElectionEvaluated()) {
+            revert ElectionNotEvaluated();
+        }
```

Comment:
Can we perform these two checks before line 322 to save reversion gas?

Reply:
Done, and please check this decision.

Added the code after the checking to an internal function that later can be implemented in mixings for the different strategies.